### ISSUE TITLE
Fix false positives for map-merge syntax in function-comma-newline-after

### ISSUE THREAD

**Issue by SimonEggert** [stylelint/stylelint#4590]
> Clearly describe the bug

I noticed false positives when using the map-merge function to add some values to an existing map. None of the three values of this rule seemed to make any difference, so I think it might be a bug in this rule.

> Which rule, if any, is the bug related to?

`function-comma-newline-after`

> What CSS is needed to reproduce the bug?

```css
$font-presets: (
  '12/12': (
    font-size: 12px,
    line-height: 1
  )
);

$font-presets: map-merge($font-presets, (
  '16/20': (
    font-size: 16px,
    line-height: 1.25
  )
));
```

> What stylelint configuration is needed to reproduce the bug?

any of
```json
{
  "rules": {
      "function-comma-newline-after": "always",
      "function-comma-newline-after": "never-multi-line",
      "function-comma-newline-after": "always-multi-line"
  }
}
```

> Which version of stylelint are you using?

`13.10.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

both CLI and VSCode Plugin

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS functions as far as I can tell.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
test.scss
8:39 Unexpected whitespace after "," in a multi-line function
```

**Comment by evilebottnawi** [comment]
It is expected, you setup ` "function-comma-newline-after": "always",`

So you need to rewrite that to:

```
$font-presets: map-merge(
  $font-presets, 
  (
    '16/20': (
      font-size: 16px,
      line-height: 1.25
    )
  )
);
```

**Comment by SimonEggert** [comment]
Thanks for replying. So basically in my example the rule will not have any effect and I should probably avoid using it altogether. Using the `--fix` option will always result in:
```scss
$font-presets: map-merge($font-presets, (
  '16/20': (
    font-size: 16px,
    line-height: 1.25
  )
));
```

Using your suggested formatting will clash with other rules I have, like
```json
  function-parentheses-newline-inside: 'never-multi-line'
  function-parentheses-space-inside: 'never
```
